### PR TITLE
feat(channel management): channel module added, the features added based on issue 66

### PR DESCRIPTION
**Pull request by Exosfeer**
Working Channel Module.
Able to 
Commands:
moveup [channelname], 
movedown [channelname], 
move [channelname] up,
move [channelname] down, 
move [channelname] up [amount],
move [channelname] down [amount],
reset [channelname]
category [channelname]
create [channelname]
delete[channelname]

### CODE REVIEW

File: Zhongli.Bot/Modules/ChannelModule.cs [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+                .WithDescription($"The channel \"{givenChannel.Name}\" has been moved downward.")
+                .WithAuthor(Context.User);
+            await Context.Channel.SendMessageAsync(embed: embed.Build());
+            SwapChannelPositions(givenChannel, 1); //Swap channel with channel above it (downward swap are positive values)
+        }
+    }
+
+    [Command("move")]
+    [Summary("Move a channel position \"up\" or \"down\" and add an integer to move more than one. Example: \"channel move [channelname] up 2\"")]
+    public async Task PositionMoveChannel(INestedChannel givenChannel, string direction, int givenNumber = 1)
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/ChannelModule.cs:188]: Use an enum instead of a string in here

File: Zhongli.Bot/Modules/ChannelModule.cs [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    {
+        // TODO: V2 Add channel creation options.
+        var channel = await Context.Guild.CreateTextChannelAsync(name);
+        var embed = new EmbedBuilder()
+           .WithTitle($"Created Channel \"{name}\" : {channel.Id}")
+           .AddField("Channel ID", channel.Id, true)
+           // .AddField() TODO: v2 Display channel creation options.
+           .WithDescription($"Successfully created channel with name \"{name}.\"")
+           .WithAuthor(Context.User);
+        await Context.Channel.SendMessageAsync(embed: embed.Build());
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/ChannelModule.cs:42]: You can use `ReplyAsync` which is shorthand for this
```suggestion
        await ReplyAsync(embed: embed.Build());
```

File: Zhongli.Bot/Modules/ChannelModule.cs [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+[RequireBotPermission(GuildPermission.ManageChannels)]
+[RequireUserPermission(GuildPermission.ManageChannels, Group = nameof(ChannelModule))]
+[RequireAuthorization(AuthorizationScope.Channels, Group = nameof(ChannelModule))]
+public class ChannelModule : ModuleBase<SocketCommandContext>
+{
+    private readonly ICommandHelpService CommandHelpService;
+    public ChannelModule(ICommandHelpService commandHelpService)
+    {
+        this.CommandHelpService = commandHelpService;
+    }
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/ChannelModule.cs:26]: Will this be used in the future?

File: Zhongli.Bot/Modules/ChannelModule.cs [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+                .WithAuthor(Context.User);
+            await Context.Channel.SendMessageAsync(embed: embed.Build());
+        }
+    }
+    /**********************************************************************************************************************/
+    /*** USEFUL PRIVATE METHODS FOR THIS MODULE ***/
+    /**
+     * Resetting all positions within the given category ID.
+     * @param ulong categoryId
+     */
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/ChannelModule.cs:305]: Should use
```suggestion
/// <summary>
/// 
/// </summary>
/// <param name="categoryId"></param>
/// <returns></returns>
```

File: Zhongli.Bot/Modules/ChannelModule.cs [hunk 5]
```diff
@@ -0,0 +1,380 @@
...
+    }
+
+
+    /* Create Channel */
+    [Command("create")]
+    [Summary("Creates a new channel.")]
+    public async Task CreateChannelAsync(string name, ChannelCreationOptions? options = null)
+    {
+        // TODO: V2 Add channel creation options.
+        var channel = await Context.Guild.CreateTextChannelAsync(name);
```
Comment (sabihoshi) [inline Zhongli.Bot/Modules/ChannelModule.cs:35]: You can do
```suggestion
        var channel = await Context.Guild.CreateTextChannelAsync(name, c => c.CategoryId = options?.ChanelCategory?.Id);
```

**Review by sabihoshi — CHANGES_REQUESTED** [review]

File: Zhongli.Bot/Modules/ChannelModule.cs [hunk 6]
```diff
@@ -0,0 +1,380 @@
...
+    {
+            await givenChannel.DeleteAsync().ContinueWith(async _ =>
+            {
+                var embed = new EmbedBuilder()
+                    .WithTitle($"Deleted Channel \"{givenChannel.Name}\" : {givenChannel.Id}")
+                    .AddField("Channel ID", givenChannel.Id, true)
+                    .WithDescription($"Disintegrating the channel from Discord databases... Don't power off your Discord.. \"{givenChannel.Name}.\"")
+                    .WithAuthor(Context.User);
+                await Context.Channel.SendMessageAsync(embed: embed.Build());
+            });
```
[the remark this answers is not in the document]
Comment (Exosfeer) [reply Zhongli.Bot/Modules/ChannelModule.cs]: True! I'll convert it into a try catch, to relay the error message!

**Comment by sabihoshi** [comment]
You can also use `{c.Mention}` instead of `\"{c.Name}\"`, I think it's much cleaner code wise as well as be better for UX

File: Zhongli.Bot/Modules/ChannelModule.cs (hunk 1 above)
Comment (Exosfeer) [reply Zhongli.Bot/Modules/ChannelModule.cs]: Converted

File: Zhongli.Bot/Modules/ChannelModule.cs (hunk 3 above)
Comment (Exosfeer) [reply Zhongli.Bot/Modules/ChannelModule.cs]: Not in this version, so we'll add it later. There will def be a version 2!

File: Zhongli.Bot/Modules/ChannelModule.cs (hunk 5 above)
Comment (Exosfeer) [reply Zhongli.Bot/Modules/ChannelModule.cs:31]: Version 2! we'll develop that! Let's do an initial release without options

**Review by sabihoshi — CHANGES_REQUESTED** [review]

**Review by sabihoshi — APPROVED** [review]

**Pull request merged**